doc/project/release_process.rst

```diff
@@ -85,24 +85,146 @@ At that point, the whole process starts over again.
+- Per Subsystem release criteria + Testing
+- Reproducible Builds
+
+Bug Severity and Priority
+=========================
+
+Priority of bugs is not enough to make release decisions and has been misused in
+the past to describe severity of an issue. Not all critical bugs are of highest
+priority and they do not need to be addressed in a release. Priority can also be
+set high on less critical bugs.
+
+Severity shall be used to classify bugs based on impact to users and community
+in general, For examples:
+
+- A kernel bug impacting every user can be set as critical or major, if it needs
+  to be fixed in the next release, it will need to be marked as a blocker bug.
+- A Kconfig option or DTS in a board is impacting a small group of users, so it
+  could be minor or moderate but, a bug in hardware that is marked as a
+  `reference hardware`` can be major or critical and in some cases a release
+  blocker.
+
+Bug priorities can be used by owners of bugs in an area or subsystem to manage
+their workflow and priorities.
+A low or moderate bug in a platform might be set to `High Priority` by the
+owner to manage execution within a subsystem or by a hardware vendor.
+
+
+Severity
+--------
+The extent to which an issue or defect can affect the software and defines the
+impact that a given defect has on the system.
+
+The following severities shall be used:
+
+- Blocker: Such an error makes it impossible to proceed with using or testing
+  the software. See below for more details.
+- Critical: bug results in a crash of the system or one or more components of
+  the system. If the failed function is unusable and
+  there is no acceptable alternative method to achieve the advertised desired
+  outcome then the severity will be stated as critical
+- Major: As with critical, but an acceptable alternative exists
+- Moderate: Causes the system to produce incorrect, incomplete or inconsistent results
+- Minor: No crashes or damage to usability of the system and desired
+  results can be easily obtained by working around the defects
+- Cosmetic: Related to look and feel of the system
+
+Severity shall be set based on the context of the bug and not in relation to the
+overall impact on the codebase. For example, a bug in an optional subsystem that
+results in a crash shall be marked as `critical` even if the bug does not impact
+every user of Zephyr.
+
+
+Blocker Bugs
+============
+
+Blocker bug process kicks during the release process and is in effect after the
+feature freeze milestone. An issue labeled as a blocker practically blocks a
+release from happening. All blocker bugs shall be resolved before a release is
+created.
 
-The current backlog of prioritized bugs shall be used as a quality metric to
-gate the final release. The following counts shall be used:
+After hard freeze (RC3), a change to the codebase can only be merged to ‘main'
+and included in the release if it fixes a bug that is granted blocker or freeze
+exception status.
```

Comment:
I would swap it around (we are in the section talking about blockers, not the section about what milestones and RC3 mean, if that makes sense)

```suggestion
A bug that is granted `blocker` status can be merged to 'main' and included in the release all the way until the hard freeze (RC3) milestone.
```